oeps/processes/oep-0056-architectural-advisory-process.rst

```diff
@@ -0,0 +1,197 @@
+
+Architectural Advisory Process (AAP)
+------------------------------------
+
+We'll create an Architecture Advisory Process to add structure to today's informal review of ADRs.  The goal of this process is to review ADRs with relevant stakeholders and experts.  Conversations in the advisory process need to take place promptly so that decisions do not feel blocked on gathering advice.
+
+We intend to also use the Advisory Process over time to revisit and iterate on our architectural principles with the goal of ensuring they're specific, measurable, achievable, realistic, and testable. (a variation of SMART criteria)  Similarly, like ADRs and architectural principles, we can use the process to review and adjust Open edX Tech Radar blips.
+
+This process is intended to be lightweight, approachable, and asynchronous.  See :ref:`principles` for more details on the spirit and functioning of the Advisory Process.
+
+Advisory Process Artifacts
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The final artifact of the Advisory Process is an Architectural Decision Record document checked in to a repository relevant to the decision.  Prior to that, however, we suggest as a best practice using Github's Issues and Pull Request tooling to conduct the Advisory Process.  The advantage of doing so is that Issue/Pull Request comments remove the burden of recording who gave particular advice, and allow us to have threaded, focused conversations about various aspects of ADR.
+
+Note that this section is GitHub-specific because that's where the Open edX platform's code resides.  The idea of "issues" and "pull requests" is otherwise portable to other version control platforms.
+
+GitHub Issues (Optional)
+************************
+
+We would encourage decision makers to use a GitHub Issue to have initial conversations around the context of a decision, or to write down upcoming/potential decisions. It may be that - at the start of a project - we're not even sure what decisions need to be made.  They may not even reveal themselves until we're in the middle of coding.
+
+An Issue is a great place to have these initial, loose, conversations about a decision space without the burden of creating a full ADR (creating a new document in the right place, authoring the sections of an ADR, creating a branch, committing, creating a PR, etc.)  It may not be necessary for more well-defined or focused decisions.  We want to preserve this as an optional first step, as we recognize that creating a "formal" ADR can feel inexplicably daunting or like a distraction, especially when possible decisions come up in the course of working on an implementation.  The mental burden of breaking "flow state" and creating a new ADR document is often problematic.  We hope jotting some notes in an Issue as a starting point will allow engineers to record their potential decision point and quickly continue their work.
```

Comment:
An issue could be a good place to have initial discussions, but new issues are not actually that discoverable.  As we adopt GitHub issues more and more, it's important to consider who will be monitoring the creation of new issues, and who will need other channels of notification that an issue has been started.

If I am considering a change to a repo that is ADR-worthy, if I make a new issue in that repo, will the right people know that the issue has been opened?

Reply:
@nedbat, do you feel this is something that should be codified here in the OEP?  My gut says that we can figure out some of the discoverability issues in our subsequent work to implement the process and that we don't need to try to specify it here before we get into iteratin' on it.  What do you think?

Reply:
I'm not sure what you are asking.  The OEP text recommends using an issue, but I have some concerns about how issues will fit into workflows.  Are you asking whether you should remove the recommendation?

Reply:
Overall, we are still working out the best communication channels, of which we have many.  Why mention issues here, but not Discourse threads?

Reply:
[clarification part of post] You mentioned that it's "important to consider who will be monitoring the creation of new issues" - I wasn't sure if you were saying you wanted some language in the OEP that talks about how that monitoring will happen; I was suggesting I didn't personally think it was necessary to put in the OEP. 😄 

[Discourse part of post] I'm honestly not sure what to think about Discourse threads and don't have a strong opinion.   Discourse feels like an alternate communication channel that we could use instead of having a conversation on GitHub (either for issues or PRs), and I'm not sure when we would choose to do that for architectural decisions.  Maybe when they're super open-ended and we want to broadcast a request for input (harkening back to your discoverability concern)?  In that case should the Discourse post be brief and encourage discussion on a GitHub Issue so we keep it all in one place?  Can we put a sentence in here that describes when to go make that Discourse broadcast?  Are there other reasons to do one or the other?